### ISSUE TITLE
False positive: `PYI015` marks assignments to `__slots__`, `__match_args__`, and `__all__` as errors

### ISSUE THREAD

**Issue by bryanforbes** [charliermarsh/ruff#4009]
Ruff version: 0.0.261

Given the following file `foo.pyi`:

```python
class One:
    __slots__ = (
        '_one',
        '_two',
        '_three',
        '_four',
        '_five',
        '_six',
        '_seven',
        '_eight',
        '_nine',
        '_ten',
        '_eleven',
    )

class Two:
    __match_args__ = (
        'one',
        'two',
        'three',
        'four',
        'five',
        'six',
        'seven',
        'eight',
        'nine',
        'ten',
        'eleven',
    )

class Three: ...
class Four: ...
class Five: ...
class Six: ...
class Seven: ...
class Eight: ...
class Nine: ...
class Ten: ...
class Eleven: ...

__all__ = (
    'One',
    'Two',
    'Three',
    'Four',
    'Five',
    'Six',
    'Seven',
    'Eight',
    'Nine',
    'Ten',
    'Eleven',
)
```

And the following invocation:

```sh
ruff --select PYI015 --isolated --no-cache foo.pyi
```

The following errors are reported:

```
foo.pyi:2:17: PYI015 [*] Only simple default values allowed for assignments
foo.pyi:17:22: PYI015 [*] Only simple default values allowed for assignments
foo.pyi:41:11: PYI015 [*] Only simple default values allowed for assignments
```

`flake8-pyi` [ignores](https://github.com/PyCQA/flake8-pyi/blob/4d575338389bbf7b27fc2dac05e833c818388431/pyi.py#L617-L622) the above assignments because they are mandatory in stub files. `ruff` should do the same.

**Comment by charliermarsh** [comment]
Planning to fix this tomorrow, along with a few other false positives.
